core/build_metadata.go

```diff
@@ -0,0 +1,40 @@
+package core
+
+import "github.com/anz-bank/sysl-go/status"
+
+var (
+	// Build metadata. These values are intended to be overridden by values
+	// supplied by the build environment at link time. Values are specified
+	// by passing "-X package.Varname=Value" to the linker. For example:
+	//
+	//  go build -ldflags="-X main.Name=BFF -X main.Version=1.0" ...etc...
```

Comment:
Could we replace this example with something more generic?
e.g BFF isn't very clear as an application name. Something like `TestApp` or `DemoApp` could be more appropriate?